src/app/util/ember-compatibility-functions.cpp

```diff
@@ -425,10 +425,29 @@ CHIP_ERROR ReadSingleClusterData(const SubjectDescriptor & aSubjectDescriptor, b
             (attributeCluster != nullptr) ? &reader : findAttributeAccessOverride(aPath.mEndpointId, aPath.mClusterId);
         if (attributeOverride)
         {
-            bool triedEncode;
-            ReturnErrorOnFailure(ReadViaAccessInterface(aSubjectDescriptor.fabricIndex, aIsFabricFiltered, aPath, aAttributeReports,
-                                                        apEncoderState, attributeOverride, &triedEncode));
-            ReturnErrorCodeIf(triedEncode, CHIP_NO_ERROR);
+            bool triedEncode = false;
```

Comment:
Why the changes here?  The logic stays the same, right?  Is it just to document the API contract for AttributeAccessInterface::Read which is already documented on that interface?

Reply:
One of my in-between changes tried to handle the fact that something is found versus not found to be supported. This goes back to optionality where an optional attribute may only be known to be optional from runtime support of some underlying data accessor. I can revert all of this now I guess, but it was *really not obvious* without a lot of reverse engineering, how the code paths on success/error yielded different outcomes.

Reply:
Removed